### DISCUSSION TITLE
How to access env variable in "contains" arguments

### DISCUSSION THREAD

**Discussion by ma-sadeghi** [community/community#25968] [Actions]
<p>Is there a way to access env variables inside <code>contains</code> arguments in an <code>if</code> statement? Something like:</p>
<pre><code class="lang-auto">name: Ubuntu

on: [push]

env:
  MSG: ${{ github.event.head_commit.message }}

jobs:
  build:
    runs-on: ubuntu-latest
    steps:
      - if: contains(${{ MSG }}, "ci skip")
        run: echo $MSG
</code></pre>

**Comment by airtower-luna** [comment]
<p>That should work using the <code>env</code> context, like <code>${{ env.MSG }}</code>.</p>

**Comment by ma-sadeghi** [comment]
<p>Here’s the error I’m getting</p>
<pre><code class="lang-auto">The workflow is not valid. .github/workflows/ci-ubuntu.yml (Line: 11, Col: 9): Unrecognized named-value: 'MSG'. Located at position 1 within expression: MSG
</code></pre>

**Comment by cschleiden** [comment]
<p>This should work, you have to use single quotes for the string - see <a href="https://help.github.com/en/actions/reference/context-and-expression-syntax-for-github-actions#literals:" rel="nofollow noopener">https://help.github.com/en/actions/reference/context-and-expression-syntax-for-github-actions#literals:</a></p>
<pre><code class="lang-auto">name: Ubuntu

on: [push]

env:
  MSG: ${{ github.event.head_commit.message }}

jobs:
  build:
    runs-on: ubuntu-latest
    steps:
      - if: contains(env.MSG, 'ci skip')
        run: echo $MSG
</code></pre>

**Comment by ma-sadeghi** [comment]
<pre><code class="lang-auto">name: Ubuntu

on: [push]

env:
  MSG: ${{ github.event.head_commit.message }}

jobs:
  build:

    runs-on: ${{ matrix.operating-system }}
    if: contains(env.MSG, 'ci skip')
</code></pre>
<p>I get the following error:<br>
<code>The workflow is not valid. .github/workflows/ci-ubuntu.yml (Line: 14, Col: 9): Unrecognized named-value: 'env'. Located at position 10 within expression: contains(env.MSG, 'ci skip')</code></p>

**Comment by BrightRan** [comment, marked as answer]
<p><a class="mention" href="/ma-sadeghi">@ma-sadeghi</a>,</p>
<p>According to the description from the docs about the <a href="https://help.github.com/en/actions/reference/context-and-expression-syntax-for-github-actions#env-context" rel="nofollow noopener"><strong>env context</strong></a>:</p>
<blockquote>
<p>You can only use the  <strong>env</strong>  context in the value of the <strong>with</strong> and <strong>name</strong> keys, or in a step’s <strong>if</strong> conditional.</p>
</blockquote>
<p>You can’t use the  <strong>env</strong>  context in a job’s <strong>if</strong> conditional.<br>
You can directly use the <a href="https://help.github.com/en/actions/reference/context-and-expression-syntax-for-github-actions#github-context" rel="nofollow noopener"><strong>github context</strong></a> in the job’s <strong>if</strong> conditional.</p>
<pre><code class="lang-yaml">jobs:
  build:
    . . .
    runs-on: ${{ matrix.operating-system }}
    if: contains(github.event.head_commit.message, 'ci skip')
</code></pre>

**Comment by tunetheweb** [comment]
<p>For anyone else stumbling on this that was as confused as I was, and going round in circle’s, the key part of “in a <strong>step’s</strong> if conditional” is the word “step”.</p>
<p>So you can use env variables in a “step’s if conditional” but not the higher level “jobs if conditional” (to skip the whole action for example).</p>
<p>Wasted far too much time on that!!</p>
